pitest-maven/src/test/java/org/pitest/maven/PitMojoTest.java

```diff
@@ -37,7 +37,7 @@ public void testRunsAMutationReportWhenMutationCoverageGoalTrigered()
     build.setOutputDirectory("foo");
     this.testee.getProject().setBuild(build);
     this.testee.execute();
-    verify(this.executionStrategy).execute(any(File.class),
```

Comment:
In line with Henry's comment on `CompoundListenerFactoryTest`, we should stick to ```any(File.class)``` and, if in doubt, use a real file object or mock. But those tests do not aim for testing the base dir parameter gets correctly handed over to the strategy. Thus I would prefer ```any(File.class)``` everywhere in this test class.
This applies to all 4 changes you've made.

Maybe we could add another test case that checks the usual case doesn't get broken. Something like

```java
  @Mock
  private MavenProject executionProject;

  @Override
  public void setUp() throws Exception {
    super.setUp();

    when(this.project.getExecutionProject()).thenReturn(executionProject);
    when(this.executionProject.getBasedir()).thenReturn(BASEDIR);
  }

  public void testUsesExecutionProjectBaseDir() throws Exception {
    this.testee = createPITMojo(createPomWithConfiguration(""));
    this.testee.execute();

    verify(this.executionStrategy).execute(eq(BASEDIR),
            any(ReportOptions.class), any(PluginServices.class), anyMap());
  }
```

Reply:
made changes.
I preserved `any(File.class)`.
To avoid `baseDir `as `null` I mocked `executionProject`